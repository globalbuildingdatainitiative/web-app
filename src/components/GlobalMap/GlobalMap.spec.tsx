import { GlobalMap } from '@components'
import { expect, render, screen, suite, test, waitFor } from '@test'
import { getGlobalMapMock, MockSessionProvider } from '@mocks'
import { MockedProvider } from '@apollo/client/testing'
import { MemoryRouter } from 'react-router-dom'

suite('GlobalMap', () => {
  const mockSessionContext = {
    loading: false as const,
    userId: '1',
    invalidClaims: [],
    doesSessionExist: true,
    accessTokenPayload: {},
  }

  test('Snapshot', () => {
    const { container } = render(
      <MockedProvider mocks={getGlobalMapMock} addTypename={false}>
        <MockSessionProvider sessionContext={mockSessionContext}>
          <MemoryRouter>
            <GlobalMap loading={false} data={undefined} />
          </MemoryRouter>
        </MockSessionProvider>
      </MockedProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  test('Render Correctly', async () => {
    render(
      <MockedProvider mocks={getGlobalMapMock} addTypename={false}>
        <MockSessionProvider sessionContext={mockSessionContext}>
          <MemoryRouter>
            <GlobalMap loading={false} data={undefined} />
          </MemoryRouter>
        </MockSessionProvider>
      </MockedProvider>,
    )
    await waitFor(() => expect(screen.getByTestId('GlobalMap')).to.exist)
  })

  // Additional tests
  test('Changes color based on filters', async () => {
    render(
      <MockedProvider mocks={getGlobalMapMock} addTypename={false}>
        <MockSessionProvider sessionContext={mockSessionContext}>
          <MemoryRouter>
            <GlobalMap loading={false} data={undefined} />
          </MemoryRouter>
        </MockSessionProvider>
      </MockedProvider>,
    )

    // Wait for data to load
    await waitFor(() => expect(screen.getByTestId('GlobalMap')).to.exist)
  })

  test('Shows error state', async () => {
    const errorMock = [
      {
        request: {
          query: getGlobalMapMock[0].request.query,
          variables: getGlobalMapMock[0].request.variables,
        },
        error: new Error('An error occurred'),
      },
    ]

    render(
      <MockedProvider mocks={errorMock} addTypename={false}>
        <MockSessionProvider sessionContext={mockSessionContext}>
          <MemoryRouter>
            <GlobalMap loading={false} data={undefined} />
          </MemoryRouter>
        </MockSessionProvider>
      </MockedProvider>,
    )

    await waitFor(() => {
      expect(screen.getByText(/Bummer/)).to.exist
    })
  })
})
