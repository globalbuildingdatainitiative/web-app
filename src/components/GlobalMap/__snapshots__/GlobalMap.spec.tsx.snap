// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GlobalMap > Snapshot 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Outfit, sans-serif;--mantine-font-family-headings: Outfit, sans-serif;--mantine-breakpoint-xxl: 120em;}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 119.99375em) {.mantine-visible-from-xxl {display: none !important;}}@media (min-width: 120em) {.mantine-hidden-from-xxl {display: none !important;}}
  </style>
  <div
    class="m_4451eb3a mantine-Center-root"
    style="height: 600px;"
  >
    <div
      class="m_4451eb3a mantine-Center-root"
      style=""
    >
      <span
        class="m_b34414df m_5ae2e3c mantine-Loader-root"
        style=""
      />
    </div>
  </div>
</div>
`;
